client/src/store/subscriptions/onUserIsCurrentUserUpdate.js

```diff
@@ -0,0 +1,22 @@
+import { requests, meAPI } from "api";
+import { entityStoreActions } from "actions";
+import isEqual from "lodash/isEqual";
+
+const { request } = entityStoreActions;
+
+export default function onUserIsCurrentUserUpdate(store) {
+  let currentState = store.getState();
+  let lastState;
+  return () => {
+    lastState = currentState;
+    currentState = store.getState();
+    const currentUser = currentState.authentication.currentUser;
+    if (!currentUser) return;
+    const lastStateUser = lastState.entityStore.entities.users[currentUser.id];
+    const currentStateUser =
+      currentState.entityStore.entities.users[currentUser.id];
+    if (!lastStateUser || !currentStateUser) return;
+    if (isEqual(lastStateUser.attributes, currentStateUser.attributes)) return;
```

Comment:
isEqual performs a deep comparison of objects, which means it's a costly execution, and in this case, this deep comparison will happen every single time the store state changes (if the state contains a user model and an authenticated user), which is going to be often. Lodash's isEqual implementation does do an inexpensive shallow check first (see https://github.com/lodash/lodash/blob/4.17.5/lodash.js#L3280), but if it didn't, you'd definitely want that shallow check as a guard in your code.